src/color.rs

```diff
@@ -126,6 +126,145 @@ pub struct Color {
+    /// use turtle::Color;
+    /// // This will not work
+    /// let color = Color::rgb(255.0, 255.0, 256.0);
+    /// ```
+    /// [`Color`]: ./index.html
+    pub fn rgb(red: f64, green: f64, blue: f64) -> Self {
+        Color::rgba(red, green, blue, 1.0)        
+    }
+
+    /// Create a new `Color` from the given RGB values and the provided alpha setting.
+    /// The given values must adhere to those laid out in the documentation for [`Color`].
+    /// Thus:
+    /// 
+    /// * 0.0 <= red <= 255.0
+    /// * 0.0 <= green <= 255.0
+    /// * 0.0 <= blue <= 255.0
+    /// * 0.0 <= alpha <= 1.0
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// let expected = Color { red: 35.0, green: 200.0, blue: 180.0, alpha: 0.5 };
+    /// let actual = Color::rgba(35.0, 200.0, 180.0, 0.5);
+    /// assert_eq!(expected, actual);
+    /// ```
+    /// 
+    /// Values that are outside the standard RGB or alpha range will result in a panic
+    /// 
+    /// ```should_panic
+    /// use turtle::Color;
+    /// // This will not work
+    /// let color = Color::rgba(255.0, 255.0, 255.0, 1.1);
+    /// ```
+    /// [`Color`]: ./index.html
+    pub fn rgba(red: f64, green: f64, blue: f64, alpha: f64) -> Self {
+        let color = Color { red, green, blue, alpha };
+        assert!(color.is_valid(), "Invalid color: {:?}. See the Color struct documentation for more information.", color);
+        color
+    }
+
+    /// Create a new `Color` from the given HSL values.
+    /// 
+    /// **NOTE:** The values given are similar to those used by [`sass`] but are only floating point values. 
+    /// This means:
+    /// 
+    /// * 0.0 <= hue <= 360.0
+    /// * 0.0 <= saturation <= 1.0
+    /// * 0.0 <= lightness <= 1.0
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// let color_rgb = Color::rgb(148.0, 139.0, 193.0);
+    /// let color_hsl = Color::hsl(250.0, 0.3, 0.65);
+    /// assert_eq!(color_rgb, color_hsl);
+    /// ```
+    /// [`sass`]: http://sass-lang.com/documentation/Sass/Script/Functions.html#hsl-instance_method
+    pub fn hsl(hue: f64, saturation: f64, lightness: f64) -> Self {
+        Color::hsla(hue, saturation, lightness, 1.0)
+    }
+
+    /// Create a new `Color` from the given HSL values and the given alpha value.
+    /// 
+    /// **NOTE:** The values given are similar to those used by [`sass`] but are only floating point values. 
+    /// This means:
+    /// 
+    /// * 0.0 <= hue <= 360.0
+    /// * 0.0 <= saturation <= 1.0
+    /// * 0.0 <= lightness <= 1.0
+    /// * 0.0 <= alpha <= 1.0
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// let color_rgba = Color::rgba(148.0, 139.0, 193.0, 1.0);
+    /// let color_hsla = Color::hsla(250.0, 0.3, 0.65, 1.0);
+    /// assert_eq!(color_rgba, color_hsla);
+    /// ```
+    /// [`sass`]: http://sass-lang.com/documentation/Sass/Script/Functions.html#hsla-instance_method
+    pub fn hsla(hue: f64, saturation: f64, lightness: f64, alpha: f64) -> Self {
```

Comment:
We should start by validating hue, saturation, lightness and alpha and then panicking with a helpful message if the values are out of range.